raft/server.cc

```diff
@@ -626,8 +635,8 @@ future<> server_impl::wait_for_entry(entry_id eid, wait_type type, seastar::abor
     }
     SCYLLA_ASSERT(inserted);
     if (as) {
-        it->second.abort = as->subscribe([it = it, &container] () noexcept {
-            it->second.done.set_exception(request_aborted());
+        it->second.abort = as->subscribe([it = it, &container, eid] () noexcept {
+            it->second.done.set_exception(request_aborted(format("Timed out waiting for idx: {}, term: {}", eid.term, eid.idx)));
```

Comment:
You should be able to access the entry index and term through `it`, to avoid copying `eid` (which happens on every call to `wait_for_entry`, not just when we abort).

Also, it doesn't have to be a timeout. It's a generic request abort, could be either due to timeout or shutdown, so I'd use the same pattern you used above for the message.